libdnf/module/module_db.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_DB_HPP
+#define LIBDNF_MODULE_MODULE_DB_HPP
+
+#include "system/state.hpp"
+
+#include "libdnf/base/base_weak.hpp"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModuleDB {
+public:
+    ModuleDB(const BaseWeakPtr & base) : base(base) {}
+
+    /// Update the `system::State` with all the changes.
+    void save_everything_to_system_state();
```

Comment:
Note - very descriptive name, but may be it is too descriptive - it is an opinion with no evidence

Reply:
I will use example - please do not take it as a suggestion for a new name.

`save_everything_to_system_state()` vs. `commit()`
The first description says not only what it does, but also where the information will be stored. In case that we will change internal implementation, it can happen that the method will have a different implementation but the name will keep original description. The longer name also is not in favor for some programmers (personal opinion in C++).

The more generic name is more robust for future changes. Then it requires only a modification of the method description, when the implementation is changed.

Reply:
Ok, you convinced me. I renamed it to `save`, but we can discuss any other name if you want.